### PR TITLE
Droth 446 viite 213 open layers 3 refactor

### PR DESCRIPTION
**Pull request by leirasf**
Pull request open to clean OTH openlayer 3 developments

### CODE REVIEW

File: UI/src/view/Styler.js [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+      //      case 99 : return 'rgba(164, 164, 162, 0.40)';
+      //    }
+      //  }
+      //} else {
+      //  if(constructionType === 1) {
+      //    return 'rgba(255, 153, 0, 0.40)';
+      //  } else {
+      //    return 'rgba(56, 56, 54, 0.40)';
+      //  }
+      //}
```
Comment (leirasf) [inline UI/src/view/Styler.js]: Remove comment code

File: UI/src/view/TileMapCollection.js [hunk 2]
```diff
@@ -1,74 +1,83 @@
...
 
-    var backgroundMapLayer = new OpenLayers.Layer.WMTS(backgroundMapConfig);
-    var aerialMapLayer = new OpenLayers.Layer.WMTS(aerialMapConfig);
-    var terrainMapLayer = new OpenLayers.Layer.WMTS(terrainMapConfig);
+    //var parser = new ol.format.WMTSCapabilities();
+    //var result = parser.read(arcgisConfig);
+    //var config = {layer: "Taustakartat_Harmaasavy"};
+    //var options = ol.source.WMTS.optionsFromCapabilities(result, config);
+    //var greyscaleLayer = new ol.layer.Tile({source: new ol.source.WMTS(options)});
+
```
Comment (leirasf) [inline UI/src/view/TileMapCollection.js]: I belive it's missing the capabilities file

File: UI/test/MultiSegmentSpeedLimitSpec.js [hunk 3]
```diff
@@ -82,15 +87,19 @@ define(['chai', 'TestHelpers'], function(chai, testHelpers) {
       before(function() {
         testHelpers.selectSpeedLimit(openLayersMap, speedLimit.id);
         backend.withSpeedLimitsData([_.flatten(speedLimitsData)]);
-        eventbus.trigger('map:moved', { selectedLayer: "speedLimit", zoom: 10, bbox: new OpenLayers.Bounds(374061, 6676946, 375292, 6678247) } );
+        //TODO check bbox
```
Comment (leirasf) [inline UI/test/MultiSegmentSpeedLimitSpec.js]: TODO can be removed

File: UI/test/SingleSegmentSpeedLimitSpec.js [hunk 4]
```diff
@@ -51,14 +52,27 @@ define(['chai', 'TestHelpers'], function(chai, testHelpers) {
         });
       });
 
+      //TODO
       describe('and clicking on the background map', function() {
```
Comment (leirasf) [inline UI/test/SingleSegmentSpeedLimitSpec.js]: TODO can be removed

File: UI/test/SpeedLimitSplitSpec.js [hunk 5]
```diff
@@ -32,14 +33,17 @@ define(['chai', 'TestHelpers'], function(chai, testHelpers) {
         openLayersMap = map;
         testHelpers.selectLayer('speedLimit');
         testHelpers.clickVisibleEditModeButton();
-        done();
+        $('.speed-limits .action.cut').click();
+        var coordinate = [0.0, 110.0];
+        testHelpers.getPixelFromCoordinateAsync(openLayersMap, coordinate, function(pixel){
+          openLayersMap.dispatchEvent({ type: 'singleclick', coordinate: coordinate, pixel: pixel });//{target: {}, srcElement: {}, xy: {x: pixel[0], y: pixel[1]}});
+          $('select.speed-limit-a option[value="100"]').prop('selected', true).change();
```
Comment (leirasf) [inline UI/test/SpeedLimitSplitSpec.js]: Comment code can be removed

File: UI/test/TestHelpers.js [hunk 6]
```diff
@@ -74,82 +74,188 @@ define(['AssetsTestData',
...
+      var layers = map.getLayers().getArray();
+      return _.find(layers, function(layer){
+          return layer.get('name') === name;
+      });
+  };
+
   var clickMarker = function(id, map) {
-    var markerBounds = _.find(map.getLayersByName('massTransitStop')[0].markers, {id: id}).bounds;
-    var markerPixelPosition = map.getPixelFromLonLat(new OpenLayers.LonLat(markerBounds.top, markerBounds.left));
+      //TODO
```
Comment (leirasf) [inline UI/test/TestHelpers.js]: All TODO's and comment code can be removed.

File: UI/test/run-integration-tests.js [hunk 7]
```diff
@@ -38,8 +38,9 @@ require(['chai',
          'GroupingInCreationSpec',
          'SingleSegmentSpeedLimitSpec',
          'SpeedLimitSplitSpec',
-         'MultiSegmentSpeedLimitSpec',
-         'BoxSelectControlSpec'],
+         'MultiSegmentSpeedLimitSpec'
+         //'BoxSelectControlSpec' WE ARE NO LONGER USING THE BOX SELECT CONTROL
```
Comment (leirasf) [inline UI/test/run-integration-tests.js]: Remove comment line and file

File: UI/src/utils/StyleRule.js [hunk 8]
```diff
@@ -0,0 +1,270 @@
...
+            var context = _.merge({}, feature.getProperties(), extraProperties);
+            var allRules = getRulesByName(name);
+            var configObj = _.merge({}, defaultStyle);
+            var returnStyles = [];
+            for(var i=0; i < allRules.length; i++){
+                var rule = allRules[i];
+                if(rule._match(context)){
+                    var styles = rule._get();
+                    for(var j=0; j < styles.length ; j++){
+                        configObj = _.merge(configObj, styles[j]);
```
Comment (leirasf) [inline UI/src/utils/StyleRule.js]: Done

File: UI/src/view/MassTransitStopLayerStyles.js [hunk 9]
```diff
@@ -0,0 +1,40 @@
+(function(root) {
+    root.MassTransitStopLayerStyles = function(roadLayer) {
+    //TODO remove the roadlayer from here because we don't need it here
```
Comment (halied) [inline UI/src/view/MassTransitStopLayerStyles.js]: Is this comment up-to-date?

**Pull request merged**